### ISSUE TITLE
Inconsistent/wrong? behavior when publishing a page with an unpublished parent

### ISSUE THREAD

**Issue by randomknowledge** [django-cms/django-cms#5634]
Hi there,

according to http://docs.django-cms.org/en/release-3.3.x/topics/publishing.html when I publish a page with an unpublished parent, the page goes into a "pending" state waiting for the parent to be published.

This works as expected, but the CMS still tries to redirect to the published version:
E.g. clicking on "publish" on a page `/unpublished-parent/child-page/?edit` sets the page to pending mode and redirects to `/unpublished-parent/child-page/?edit_off=true` wich leads to a 404 obviously.

Also from within the page tree there's no possibility to publish/set to pending. This is inconsistend in my opinion.

All the best

Florian

**Comment by evildmp** [comment]
I think I agree that the behaviour when you hit **Publish** on the child on an unpublished page is incorrect.

The question is, where should the system redirect the user? I think it should probably stay on the same page, but remain in edit mode.

For your other point, *Pending* just means *Published (but waiting for its parent to be published)*. It's not something you can apply yourself to a page.

**Comment by randomknowledge** [comment]
Thanks four your reply, @evildmp!
I think staying in edit mode is a good idea - maybe in combination with a notification?
Regarding the other point - I know that Pending just means Published, but it's simply not possible to publish a page with an unpublished parent from within the admin page tree (the "publish" button is missing).

Thanks and all the best
Florian

**Comment by czpython** [comment]
I think we should hide the Publish button from a child page if it's parent is unpublished.
To keep consistency with the behavior in the page list.

And like in the page list, if someone manually goes to the `/publish` url then we set the pending state, redirect to edit mode and display the message:

`Page not published! A parent page is not published yet.`

A more complete fix in my opinion would be to not set the page as pending when a user tries to publish it and instead inform the user immediately that the page can't be published. This means that the only time a page is set as pending is when moved under an unpublished page.

**Comment by mkoistinen** [comment]
Agreed. This will also help educate users about unpublished parent pages 'hiding' children pages.

**Comment by yakky** [comment]
:+1:

**Comment by victor-yunenko** [comment]
I think there already was a warning that was asking the user to confirm that this action is going to publish all the parent pages?

In any case that looks to me as something that the user would want to do in this situation.

**Comment by fsbraun** [comment]
The behavior haas changed with django CMS 4 and there is no pending state any more,
